editor/src/messages/tool/tool_messages/select_tool.rs

```diff
@@ -1143,9 +1157,12 @@ impl Fsm for SelectToolFsmState {
 				responses.add(FrontendMessage::UpdateInputHints { hint_data });
 			}
 			SelectToolFsmState::DrawingBox { .. } => {
-				// TODO: Add hint and implement functionality for holding Shift to extend the selection, thus preventing the prior selection from being cleared
-				// TODO: Also fix the current functionality so canceling the box select doesn't clear the prior selection
-				let hint_data = HintData(vec![HintGroup(vec![HintInfo::mouse(MouseMotion::Rmb, ""), HintInfo::keys([Key::Escape], "Cancel").prepend_slash()])]);
+				// TODO: Re-select deselected layers during drag when Shift is pressed, and re-deselect if Shift is released before drag ends.
```

Comment:
This can be moved to the line above the commented out `HintGroup...` line. I also recommend linking to `https://discord.com/channels/731730685944922173/1216976541947531264/1321360311298818048` for context to the future reader of this comment.